### PR TITLE
Add check on backend being up before frontend command

### PR DESCRIPTION
**Pull request by m2web**
### Motivation and Context
I noted that when running the Start.ps1 the backend would not be running via `Start-Process pwsh -ArgumentList "-command $BackendScript"` before the frontend command `& $FrontendScript`. I added a condition to check that the backend is running before starting the frontend process.

### Description
I added a condition to check that the backend is running before starting the frontend process.

Other changes:
I update the comment in the /scripts/start.sh to be like /scripts/Start.ps1 file

### Contribution Checklist

- [x ] The code builds clean without any errors or warnings
- [x ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [x ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by teresaqhoang** [comment]
Hey @m2web, it looks like you're creating a new branch and pull request for each change you're making. Are you intentionally keeping trying to keep these changes isolated?

If not, I would recommend pushing your changes to one branch. You can do this by:

1. `git checkout main` and `git pull` to ensure you have the latest changes in main.
2. `git checkout -b unique-branch-name`
3. Make some changes.
4. `git commit -am "Some message that's descriptive to the changes you made"`
     - `a` in -am stands for "Add all changed files"
     - `m` stands for "Include commit message" 
5. `git push`

Steps 1 and 2 only need to be done once.
You can repeat steps 3-5 anytime you make a change, and when you think you're done with all your changes, you can create a pull request for us to review. 
A quick way to create a pull request is to navigate directly using a URL unique to your branch name (replace the end):
`https://github.com/microsoft/chat-copilot/compare/main...unique-branch-name`

**Comment by teresaqhoang** [comment]
Closing https://github.com/microsoft/chat-copilot/pull/311/commits as a dupe of this PR. 

Please add the change that removes the webapi/appsettings.Development.json to this branch by running the following commands:

1. `git checkout update-comment-in-start.sh`
1. Remove webapi/appsettings.Development.json
1. `git commit -am "Removing appsettings.development.json"`
1. `git push`

**Comment by m2web** [comment]
teresaqhoang, Yes, I like to keep the content of a branch as singular and clean as is warranted by the context. Sorry for the confusion with the various branches. I have used git for years although you would not know it given these commits 🙂

File: scripts/Start.ps1 [hunk 1]
```diff
@@ -8,6 +8,20 @@ $FrontendScript = Join-Path "$PSScriptRoot" 'Start-Frontend.ps1'
 
 # Start backend (in new PS process)
 Start-Process pwsh -ArgumentList "-command $BackendScript"
+# check if the backend is running before proceeding
+$backendRunning = $false
+
+# get the port from the REACT_APP_BACKEND_URI env variable
+$envFilePath = Join-Path $PSScriptRoot '..\webapp\.env'
+$envContent = Get-Content -Path $envFilePath
+# Write-Output $envContent
```
Comment (glahaye) [inline scripts/Start.ps1:17]: Comments at lines 17 and 19 should be removed. Please do so in another commit, not another branch! ;)

File: scripts/start.sh [hunk 2]
```diff
@@ -7,8 +7,19 @@ set -e
 ScriptDir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 cd "$ScriptDir"
 
+# get the port from the REACT_APP_BACKEND_URI env variable
+envContent=$(grep -v '^#' ../webapp/.env | xargs)
+backendPort=$(echo $envContent | sed -n 's/.*:\([0-9]*\).*/\1/p')
+# echo "backendPort: $backendPort"
```
Comment (glahaye) [inline scripts/start.sh:13]: This comment should be removed

**Review by glahaye — APPROVED** [review]
LGTM

File: scripts/Start.ps1 [hunk 3]
```diff
@@ -8,6 +8,18 @@ $FrontendScript = Join-Path "$PSScriptRoot" 'Start-Frontend.ps1'
...
 Start-Process pwsh -ArgumentList "-command $BackendScript"
+# check if the backend is running before proceeding
+$backendRunning = $false
+
+# get the port from the REACT_APP_BACKEND_URI env variable
+$envFilePath = Join-Path $PSScriptRoot '..\webapp\.env'
+$envContent = Get-Content -Path $envFilePath
+$port = [regex]::Match($envContent, ':(\d+)/').Groups[1].Value
+
+while ($backendRunning -eq $false) {
```
Comment (teresaqhoang) [inline scripts/Start.ps1:19]: Would be a good idea to add a timeout in case the backend server goes dark for any reason

File: scripts/start.sh [hunk 4]
```diff
@@ -7,8 +7,18 @@ set -e
...
 
+# get the port from the REACT_APP_BACKEND_URI env variable
+envContent=$(grep -v '^#' ../webapp/.env | xargs)
+backendPort=$(echo $envContent | sed -n 's/.*:\([0-9]*\).*/\1/p')
+
 # Start backend (in background)
 ./start-backend.sh &
 
+# check that the backend is running and keep checking until it is
+while ! nc -z localhost $backendPort; do
```
Comment (teresaqhoang) [inline scripts/start.sh:18]: Same here - would be a good idea to add a timeout in case the backend server goes dark for any reason

**Review by teresaqhoang — COMMENTED** [review]
Updated the PR descriptions to align with your original intent! Thanks for this change!

**Comment by teresaqhoang** [comment]
> teresaqhoang, Yes, I like to keep the content of a branch as singular and clean as is warranted by the context. Sorry for the confusion with the various branches. I have used git for years although you would not know it given these commits 🙂

Ah in general, our PR philosophy is any comments left on a PR can be addressed in the same branch, unless it's a different feature request or domain. Sorry if this was unclear!

File: scripts/Start.ps1 (hunk 3 above)
Comment (m2web) [reply scripts/Start.ps1:19]: Agreed. I just found the SK Office Hours on YouTube. Very helpful! Also, the Copilot series with Alex Chao ( @alexchaomander ) is very informative as well. 🙂

Comment (teresaqhoang) [reply scripts/Start.ps1:19]: Super happy to hear you found it helpful :) You're welcome to OH anytime, we host them every Wednesday.

**Comment by teresaqhoang** [comment]
Closing as dupe of https://github.com/microsoft/chat-copilot/pull/319
